game_patch/multi/server.cpp

```diff
@@ -327,29 +327,24 @@ const char* get_rand_level_filename()
     std::size_t num_levels = rf::netgame.levels.size();
 
     // current_level_index keeps our position in rotation while we go somewhere else
-    const char* current_level_filename = rf::level_filename_to_load;
-
-    xlog::debug("Starting random selection process, current level is {}", current_level_filename);
+    rf::String filename = rf::level_filename_to_load;
 
     if (num_levels <= 1) {
         // nowhere else to go, we're staying here!
-        xlog::debug("Map list has only 1 member, reloading current level {}", current_level_filename);
-        return current_level_filename;
+        return filename.c_str();
     }
 
     std::uniform_int_distribution<std::size_t> dist_levels(0, num_levels - 1);
     std::size_t rand_level_index = dist_levels(rng);
 
     // keep rolling until we get something that's not the current level
-    while (rf::netgame.levels[rand_level_index] == current_level_filename) {
+    while (rf::netgame.levels[rand_level_index] == filename) {
         rand_level_index = dist_levels(rng);
     }
```

Comment:
What if rotation is `[dm02, dm02]`?

Reply:
Thanks for this, I hadn't accounted for that case.

I just pushed a change to resolve this. Technically the newest iteration does make it twice as likely for one specific level to be selected each time it is run, but I don't see that as a problem especially given the one that is twice as likely to be selected is randomly chosen each time.